### ISSUE TITLE
non-exhaustive patterns: `Mesh` not covered

### ISSUE THREAD

**Issue by LinusU** [gtk-rs/glib#61]
I'm getting this error when trying to build glib (I'm trying to do `cargo doc` for `cairo-rs`).

```text
   Compiling glib v0.0.3 (https://github.com/gtk-rs/glib#607030ed)
src/patterns.rs:33:5: 42:6 error: non-exhaustive patterns: `Mesh` not covered [E0004]
src/patterns.rs:33     match pattern_type {
src/patterns.rs:34         PatternType::Solid            => Box::new(SolidPattern::wrap(ptr))   as Box<Pattern>,
src/patterns.rs:35         PatternType::Surface          => Box::new(SurfacePattern::wrap(ptr)) as Box<Pattern>,
src/patterns.rs:36         PatternType::LinearGradient   => Box::new(LinearGradient::wrap(ptr)) as Box<Pattern>,
src/patterns.rs:37         PatternType::RadialGradient   => Box::new(RadialGradient::wrap(ptr)) as Box<Pattern>,
src/patterns.rs:38         #[cfg(cairo_1_12)]
```

**Comment by gkoz** [comment]
The error comes from cairo, actually. It's a cargo issue I haven't gotten around to tackle: `cargo` doesn't pass `cfg`s returned by the build script to `rustdoc`.

**Comment by gkoz** [comment]
Thanks for reporting the issue though.

**Comment by gkoz** [comment]
This is fixed in `cargo 0.6.0 2015-10-12` and later. `cargo doc` from the recent nightly no longer fails.

**Comment by LinusU** [comment]
Cool, nice work :tada:
